### PR TITLE
CI: Install pyarrow-core instead of pyarrow from conda-forge

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Reduce unnecessary optional dependencies when installing `pyarrow` from conda-forge. Xref https://github.com/conda-forge/pyarrow-feedstock/pull/111

From https://github.com/conda-forge/pyarrow-feedstock/blob/381a88b6f8957204e34047151c4ca8a3f1e54204/recipe/meta.yaml#L157-L164, this means we won't be installing `libarrow-acero`, `libarrow-dataset`, `libarrow-substrait` and `libparquet`.

Fixes #

**Preview**:

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

File: .github/workflows/ci_tests.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
             numpy-version: '1.25'
             pandas-version: '=2.0'
             xarray-version: '=2023.04'
-            optional-packages: ' contextily geopandas<1 ipython pyarrow rioxarray sphinx-gallery'
+            optional-packages: ' contextily geopandas<1 ipython pyarrow-core rioxarray sphinx-gallery'
```
Comment (seisman) [inline .github/workflows/ci_tests.yaml:76]: Also need to update line 82.

Do we also want to update `doc/install.md`?

Comment (weiji14) [reply .github/workflows/ci_tests.yaml:76]: Good catch, done in commits 3e59046581ba796912b0a475391d17bdce6d9d82 and 0cef357ae4a76e95456c7166d442d59ec5c48e3b

**Comment by seisman** [comment]
Aslo need to note that pyarrow-core is available since pyarrow v16 (released in April, 2024). Installing `pyarrow-core` in `ci_tests_legacy.yaml` may cause issues in https://github.com/GenericMappingTools/pygmt/pull/3639#discussion_r1853425363.

**Comment by weiji14** [comment]
> Aslo need to note that pyarrow-core is available since pyarrow v16 (released in April, 2024). Installing `pyarrow-core` in `ci_tests_legacy.yaml` may cause issues in [#3639 (comment)](https://github.com/GenericMappingTools/pygmt/pull/3639#discussion_r1853425363).

Yes, thanks for the reminder, I think we can use `pyarrow-core` here, and I'll revert to using `pyarrow` in #3639. That PR is still a little experimental.

**Review by seisman — APPROVED** [review]

**Pull request merged**